### DISCUSSION TITLE
magit on startup

### DISCUSSION THREAD

**Discussion by ivfisunov** [syl20bnr/spacemacs#16218] [Q&A] [answered]
Is there a way to configure Magit to open git status buffer (SPC g s) on spacemacs startup?
Thanks in advance.

**Comment by alexey0308** [comment, marked as answer]
Hi Ivan @ivfisunov, as I understand you would like to have a magit buffer opened for a specific repository on startup - does it work for you if you add to your `user-config` something like

```elisp
   (progn 
    (magit-status-setup-buffer "~/path/to/repo-name")
    (switch-to-buffer (get-buffer "magit: repo-name"))
    (delete-other-windows))
```
?

**Reply by pataquets** [reply]
If you're using `projectile` I've also found useful to use this:
``` elisp
(setq projectile-switch-project-action 'projectile-vc)
```
See [docs](https://docs.projectile.mx/projectile/configuration.html#switching-projects)
